core/src/main/java/com/google/cloud/sql/core/CloudSqlInstance.java

```diff
@@ -53,8 +54,8 @@
   private final KeyPair keyPair;
 
   private final Object instanceDataGuard = new Object();
-  private ListenableFuture<InstanceData> currentInstanceData;
-  private ListenableFuture<ListenableFuture<InstanceData>> nextInstanceData;
+  private volatile ListenableFuture<InstanceData> currentInstanceData;
```

Comment:
I'm no longer convinced these need to be volatile since they're covered by synchronized blocks.

Reply:
But consider annotating both with [`@GuardedBy("instanceDataGuard")`](https://github.com/google/error-prone/blob/master/annotations/src/main/java/com/google/errorprone/annotations/concurrent/GuardedBy.java). That should help you avoid unsynchronized access.